### PR TITLE
feat: Update auth flow

### PR DESCRIPTION
**Pull request by cyaiox**
This PR updates the auth flow to use the new `auth dapp`.

This change required updating the signing entities process flow and how to store the signatures.

It also updates the bundler process to use `vite` and upgrade dependencies.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    />
+    <link rel="apple-touch-icon" href="%VITE_BASE_URL%/logo192.png" />
+    <link rel="manifest" href="%VITE_BASE_URL%/manifest.json" />
+    <title>Linker App</title>
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <script type="module" src="/src/index.tsx"></script>
+    <script type="text/javascript">
```
Comment (gonpombo8) [inline index.html:20]: whats this script here ?

File: src/components/IdentityPage/IdentityPage.tsx [hunk 2]
```diff
@@ -1,53 +1,45 @@
...
-      {isModalOpen ? <LoginModal
-          name='IdentityPageLoginModal'
-          onClose={() => setIsModalOpen(false)}
-          onConnect={onConnectWallet}
-          isLoading={isConnecting}
-        /> : null}
     </div>
-  )
+  ) : null
 }
```
Comment (cyaiox) [inline src/components/IdentityPage/IdentityPage.tsx:45]: is this component still required?

File: src/config/env/prod.json [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-  "CHAIN_ID": "1"
-}
\ No newline at end of file
+  "CHAIN_ID": "1",
+  "ENVIRONMENT": "production",
+  "AUTH_URL": "/auth",
+  "SEGMENT_API_KEY": "sFdziRVDJo0taOnGzTZwafEL9nLIANZ3"
```
Comment (gonpombo8) [inline src/config/env/prod.json:5]: can we pass this variables via the github actions ? I know they are already public but being exposed here feels weird.
Also why we need segment for this ? What if the user doesnt want to be tracked like we give the option in the sdk-commands ?

File: src/config/env/prod.json [hunk 4]
```diff
@@ -1,3 +1,6 @@
 {
-  "CHAIN_ID": "1"
-}
\ No newline at end of file
+  "CHAIN_ID": "1",
+  "ENVIRONMENT": "production",
+  "AUTH_URL": "/auth",
```
Comment (gonpombo8) [inline src/config/env/prod.json:4]: this shouldn't be `/auth/login` ? 
Not seeing this route on the server

**Review by gonpombo8 — APPROVED** [review]

**Comment by cazala** [comment]
I'm seeing this error when i try to run the app:

Screenshot 2024-09-10 at 4 32 55 PM

File: src/config/env/prod.json (hunk 3 above)
Comment (cyaiox) [reply src/config/env/prod.json:5]: Actually, `segment` is not required in the latest change. I'll remove it! thanks for the feedback

File: src/config/env/prod.json (hunk 4 above)
Comment (cyaiox) [reply src/config/env/prod.json:4]: the `auth dApp` works under the `/auth` path, the `login` endpoint is concatenated when required in the login flow

**Comment by cyaiox** [comment]
> I'm seeing this error when i try to run the app:
> 
> Screenshot 2024-09-10 at 4 32 55 PM

@cazala could you please try again? Added the logic to load the file .env.default when the file .env is not present

File: src/components/IdentityPage/IdentityPage.tsx (hunk 2 above)
Comment (gonpombo8) [reply src/components/IdentityPage/IdentityPage.tsx:45]: not sure when we were using this..
for sure it is not being used for the deploy command

**Comment by cazala** [comment]
LGTM!


Screenshot 2024-09-11 at 2 14 55 PM

**Review by cazala — APPROVED** [review]

**Pull request merged**